### PR TITLE
Crop harvest and replant

### PR DESCRIPTION
**Pull request by broman**
Implements #3. Tested and working as expected!

### CODE REVIEW

File: src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          })
+        }
+      }
+    }
+  }
+
+  def isCrop(item: Material): Boolean = {
+    val crops = List(Material.WHEAT, Material.POTATOES, Material.CARROTS, Material.BEETROOTS)
+    crops.contains(item)
+  }
```
Comment (xJade00) [inline src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala:49]: Creating a list like this? Ehhhh. I'd use a match.
```scala
def isCrop(item: Material): Boolean = item match {
    case _ @ (Material.WHEAT | Material.POTATOES | Materials.CARROTS | Material.BEETROOTS) => true
    case _ => false
}
```

Also cleaner.

File: src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    val location = block.getLocation
+    val factionAtBlock = Board.getInstance.getFactionAt(new FLocation(location))
+    if(faction.equals(factionAtBlock) || factionAtBlock.isWilderness) {
+      // If the faction is player's faction, or the faction is the wild, we can replant
+      val material = block.getType
+      if(isCrop(material)) {
+        val blockAsAgeable = block.getBlockData.asInstanceOf[Ageable]
+        if(blockAsAgeable.getAge == blockAsAgeable.getMaximumAge) {
+          // If the interacted block is a full crop, we harvest
+          val itemsToDrop = getApplicableDrop(material).filter(_.getAmount != 0)
```
Comment (xJade00) [inline src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala]: Remove the val assignment and just do the foreach straight on it.

File: src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    val fPlayer = FPlayers.getInstance.getByPlayer(event.getPlayer)
+    val faction = fPlayer.getFaction
+    val location = block.getLocation
+    val factionAtBlock = Board.getInstance.getFactionAt(new FLocation(location))
+    if(faction.equals(factionAtBlock) || factionAtBlock.isWilderness) {
+      // If the faction is player's faction, or the faction is the wild, we can replant
+      val material = block.getType
+      if(isCrop(material)) {
+        val blockAsAgeable = block.getBlockData.asInstanceOf[Ageable]
```
Comment (xJade00) [inline src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala]: This breaks conventions. Use pattern matching, not casting.

**Review by xJade00 — CHANGES_REQUESTED** [review]

File: src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala (hunk 2 above)
Comment (broman) [reply src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala]: Readability

File: src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala (hunk 3 above)
Comment (broman) [reply src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala]: Done

File: src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala (hunk 2 above)
Comment (broman) [reply src/main/scala/com/mewna/mc/box/listeners/CropHarvestListener.scala]: Done

**Pull request merged**